src/sqlfluff/core/parser/lexer.py

```diff
@@ -371,6 +394,27 @@ def elements_to_segments(
                     elif source_only_slice.source_idx >= source_slice.start:
                         so_slices.append(source_only_slice)
 
+                # TODO: I don't think this code will generate substitution_start
+                # and substitution_end if a templated block evaluates to the empty
+                # string. Is this actually possible, and if so, is it an issue?
```

Comment:
Curious to see this for `{{ config(materialized='view') }}` in dbt as thats the main example of a block that evaluates to an empty string and I have a feeling it gets excluded from the source slices

Reply:
👍 I'll try that now. I tried some of my own blank macros or values, but testing the real dbt thing will be the 🥇 standard.

Reply:
The dbt test works, too. 🎉🥳

My intuition is that it's _possible_ for a substitution to return a blank value, but I have yet to figure out _how_.

Parse output:
```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    [META] placeholder:                                       [Type: 'substitution_start', Raw: "{{ config(materialized='view') }}"]
[L:  1, P:  1]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    [META] placeholder:                                       [Type: 'substitution_end', Raw: ' ']
[L:  2, P:  1]      |    newline:                                                  '\n'
[L:  3, P:  1]      |    statement:
[L:  3, P:  1]      |        select_statement:
[L:  3, P:  1]      |            select_clause:
[L:  3, P:  1]      |                keyword:                                      'SELECT'
[L:  3, P:  7]      |                [META] indent:
[L:  3, P:  7]      |                whitespace:                                   ' '
[L:  3, P:  8]      |                select_clause_element:
[L:  3, P:  8]      |                    wildcard_expression:
[L:  3, P:  8]      |                        wildcard_identifier:
[L:  3, P:  8]      |                            star:                             '*'
[L:  3, P:  9]      |            whitespace:                                       ' '
[L:  3, P: 10]      |            [META] dedent:
[L:  3, P: 10]      |            from_clause:
[L:  3, P: 10]      |                keyword:                                      'FROM'
[L:  3, P: 14]      |                whitespace:                                   ' '
[L:  3, P: 15]      |                from_expression:
[L:  3, P: 15]      |                    [META] indent:
[L:  3, P: 15]      |                    from_expression_element:
[L:  3, P: 15]      |                        table_expression:
[L:  3, P: 15]      |                            table_reference:
[L:  3, P: 15]      |                                identifier:                   'foo'
[L:  3, P: 18]      |                    [META] dedent:
[L:  3, P: 18]      |    newline:                                                  '\n'
```

Reply:
Nice, that's pretty promising!